core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -5914,6 +5866,73 @@ enum DefaultConstants NADefaults::validateAndInsert(const char *attrName,
 
 } // NADefaults::validateAndInsert()
 
+float NADefaults::computeNumESPsPerCore(NABoolean aggressive)
+{
+   #define DEFAULT_ESPS_PER_NODE 2   // for conservation allocation
+   #define DEFAULT_ESPS_PER_CORE 0.5 // for aggressive allocation
+
+     // Make sure the gpClusterInfo points at an NAClusterLinux object.
+     // In osim simulation mode, the pointer can point at a NAClusterNSK
+     // object, for which the method numTSEsForPOS() is not defined.
+   NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
+   assert(gpLinux);		
+
+   // cores per node
+   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
+
+   if ( aggressive ) {
+      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
```

Comment:
My comment is more that right now it's hard-coded to non-aggressive as far as I can see. Don't you want to have a CQD to turn it on and off - since you added a CQD. But, as far as I can see, the new CQD does nothing right now?